Kernel/FileSystem/BlockBasedFileSystem.h

```diff
@@ -29,8 +29,15 @@
 #include <Kernel/FileSystem/FileBackedFileSystem.h>
 
 namespace Kernel {
+struct CacheEntry {
```

Comment:
Please put this inside the `BlockBasedFS` class so we don't add a `CacheEntry` class in the `Kernel` namespace that everyone can see. :)